pkg/kubelet/config/flags.go

```diff
@@ -88,6 +91,7 @@ func (s *ContainerRuntimeOptions) AddFlags(fs *pflag.FlagSet) {
 	// General settings.
 	fs.StringVar(&s.ContainerRuntime, "container-runtime", s.ContainerRuntime, "The container runtime to use. Possible values: 'docker', 'remote'.")
 	fs.StringVar(&s.RuntimeCgroups, "runtime-cgroups", s.RuntimeCgroups, "Optional absolute name of cgroups to create and run the runtime in.")
+	fs.StringVar(&s.CRIVersion, "cri-version", s.CRIVersion, fmt.Sprintf("Specify the CRI version for the remote runtime and image endpoint to be used. Can be either empty (chosen automatically on connection) or forced to %q and %q (deprecated)", remote.APIVersionV1, remote.APIVersionV1alpha2))
```

Comment:
It looks like none of these flags are configurable via the [config API](https://kubernetes.io/docs/reference/config-api/kubelet-config.v1beta1). The majority of these ContainerRuntimeOptions will be deprecated/removed upon Docker removal.

I would expect that we validate the versions (we don't appear to be doing so right now?) and that we consider adding CRIVersion and ContainerRuntime to the config API (since imo it doesn't make sense to set CRIVersion if ContainerRuntime=docker).

Reply:
> It looks like none of these flags are configurable via the [config API](https://kubernetes.io/docs/reference/config-api/kubelet-config.v1beta1). The majority of these ContainerRuntimeOptions will be deprecated/removed upon Docker removal.

Good, added the `CRIVersion` option to the configuration API, whereas the CLI flag will be treated with a higher priority.
 
> I would expect that we validate the versions (we don't appear to be doing so right now?) and that we consider adding CRIVersion and ContainerRuntime to the config API (since imo it doesn't make sense to set CRIVersion if ContainerRuntime=docker).

We validate the CRI version in the remote runtime determination: https://github.com/kubernetes/kubernetes/pull/106006/files#diff-1a7a9a00739661ddc6e3dd02d6552b6e83240436ec7bc69e1e9b6bce6eb44caeR109-R137

Should I add the `ContainerRuntime` options within this PR as well or do we wanna defer it to another follow-up PR?

Reply:
Do we not have any validation for the CLI flags? We do for the config APIs... https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/apis/config/validation/validation.go

Reply:
Sure, added the validation to `cmd/kubelet/app/options/options.go` and `pkg/kubelet/apis/config/validation/validation.go`